core/protos/o2control.proto

```diff
@@ -276,11 +276,17 @@ message GetRolesReply {
     repeated RoleInfo roles = 1;
 }
 
-message GetWorkflowTemplatesRequest{}
+message GetWorkflowTemplatesRequest{
+    string repoPattern = 1;
+    string revisionPattern = 2;
+    bool allBranches = 3;
+    bool allTags = 4;
+}
 
 message WorkflowTemplateInfo {
     string repo = 1;
-    string template = 2;
+    string revision = 2;
+    string template = 3;
```

Comment:
Can we do this without reindexing fields?
```
string template = 2;
string revision = 3;
```
should work just as well, and it would improve `proto` interface compatibility.